src/esperanza/validatorstate.h

```diff
@@ -17,56 +17,71 @@
 namespace esperanza {
 
 // clang-format off
+// All phases are ordered in a way how they can progress.
+// Every phase changes either one step up and down (if re-org happens).
+// Every phase has sparse index because if we want to introduce
+// a new one, we can include it in between without breaking
+// a layout on disk
 BETTER_ENUM(
     _Phase,
     uint8_t,
-    NOT_VALIDATING,
 
-    // UNIT-E TODO: replace IS_VALIDATING in favor of state->IsFinalizerVoting
-    // as since BlockConnected is processed on a thread, this phase can be
-    // set with the delay. Another issue is that AddToWalletIfInvolvingMe
-    // is called before BlockConnected so the state is lagging one block behind
-    IS_VALIDATING,
+    // finalizer didn't send deposit
+    NOT_VALIDATING = 10,
 
-    WAITING_DEPOSIT_CONFIRMATION,
-    WAITING_DEPOSIT_FINALIZATION
+    // deposit is in the mempool/wallet but is not included in a block
+    WAITING_DEPOSIT_CONFIRMATION = 20,
+
+    // deposit is included in a block but an epoch of this block
+    // is not finalized yet
+    WAITING_DEPOSIT_FINALIZATION = 30,
```

Comment:
that's correct but it kept it as it says `FINALIZATION` and not `FINALIZED` so it's more or less aligned. Which name do you propose?

Reply:
WAITING_START_DYNASTY?

Reply:
@Gnappuraz could you, please, share your opinion about renaming `WAITING_DEPOSIT_FINALIZATION` to `WAITING_START_DYNASTY`?

I don't mind to put more effort and change this phase too (notice it wasn't introduced in this PR) but would like to agree on this since it requires changes in many places (unit/function tests too) so would like to do it once.

Reply:
I'm fine if you keep the name of enum in this PR, but would be great if you adjust the comment at least.

Reply:
Corrected the comment